### ISSUE TITLE
`issubclass` doesn't narrow metaclasses

### ISSUE THREAD

**Issue by DetachHead** [python/mypy#11671]
```py
class FooMetaclass(type):
    pass
class Foo(metaclass=FooMetaclass):
    pass

fm: FooMetaclass
reveal_type(fm)  # note: Revealed type is "__main__.FooMetaclass"
if issubclass(fm, Foo):
    reveal_type(fm)  # note: Revealed type is "__main__.FooMetaclass"
```
https://mypy-play.net/?mypy=master&python=3.10&gist=d03a0fb47ada04cb3e352ed13fc0f58b

**Comment by sobolevn** [comment]
Also related:

```python
class A:
    pass

class B:
    pass

t = A
if issubclass(t, B):
    reveal_type(t)  # Should be reported as unreachable, because `t` is never a subtype of `B`
```

But, this reveals: `Revealed type is "def () -> ex.A"`

I will going to look into it! Thanks!

**Comment by sobolevn** [comment]
Maybe this will make more sense?

```python
class A:
    x: int

class B:
    y: str

t: type[A]
if issubclass(t, B):
    reveal_type(t)  # N: Revealed type is "Type[ex.<subclass of "A" and "B">]"
    reveal_type(t.x)  # N: Revealed type is "builtins.int"
    reveal_type(t.y)  # N: Revealed type is "builtins.str"
```

**Comment by montag451** [comment]
Any progress on this issue ?

**Comment by sobolevn** [comment]
Thanks for the ping, I've sent a PR.
